### ISSUE TITLE
Grid does not render properly when it was initially hidden

### ISSUE THREAD

**Issue by AxelBeck** [angular-ui/ng-grid#3108]
I think it's the same issue as descriped and fixed for Version 1.3.8 here: https://github.com/angular-ui/ng-grid/pull/52

I am using:
     angular v1.3.8
     ui-grid v3.0.0-rc.20

and can reproduce this in
     IE             v 11.0.9600.17691
     Chrome    v 41.0.2272.101 m
     Firefox     v 36.0.4

**Comment by PaulL1** [comment]
Refer http://ui-grid.info/docs/#/tutorial/108_hidden_grids.  Is it a defect or expected behaviour?

**Comment by AxelBeck** [comment]
Setting the width from 100% to an px value works.
The "ng-if Trick" does not work for me. Perhaps the problem is, that I use the grid inside of my own directive.
I will try to reproduce this in plunker soon

**Comment by AxelBeck** [comment]
Here the promised plunker: http://plnkr.co/edit/7DhisDMJPePkOZp1Zf6z?p=preview

**Comment by yurigenin** [comment]
You can fix it by using uiGridAutoResize feature but it may cause other issues like flickering on resize. See my other issues for a plunker example.

**Comment by yurigenin** [comment]
By the way, this is again caused by the selection header (any column pinned to the left) being present. The body (central) container is a static div so it always bumps against the left pinned container div and forced down by the layout engine. This is what you see. All container ideally should use absolute positioning so that browser does not 'bully' them around.

**Comment by noherczeg** [comment]
We are using ui-grid with the selection header, but without the pinning feature. I managed to do a workaround with:

- multiple grids (some of which are hidden initially)
- selection header
- auro-resize on

Without:
- flickering
- misplacement

```
.ui-grid-render-container-body {
    position: absolute;
    margin-left: 31px;
}

.ui-grid-grid-footer {
    position: absolute;
    bottom: 25px;
}
```

Keep in mind, that the pinning feature basically breaks this way, because of the CSS override.

This will be fine (for us) until someone comes up with a better alternative / fix for this.

**Comment by c0bra** [comment]
This should be fixed with #3371. Please file another issue if something's still not right for you. I've validated that the plunkers appear to be working well.

**Comment by SuprasannaBhaumik** [comment]
The rendering of the next row elements have by default used "float:left" and hence due to the available space the columns fit into that. 
Changing this to clear:both will ensure that the new rows are rendered in the next block.
Add this rule to the ui-grid-row class
.ui-grid-row{
clear:both;
}
